packages/client/components/AddTeamDialog.tsx

```diff
@@ -0,0 +1,201 @@
+  const fieldsetStyles = `mx-0 mb-6 flex flex-col w-full p-0`
+
+  const [selectedUsers, setSelectedUsers] = React.useState<Option[]>([])
+  const [mutualOrgsIds, setMutualOrgsIds] = React.useState<string[]>([])
+
+  const showOrgPicker = !!(
+    selectedUsers.length &&
+    (mutualOrgsIds.length > 1 || !mutualOrgsIds.length)
+  )
+
+  const defaultOrgId = mutualOrgsIds[0]
+  const [selectedOrgId, setSelectedOrgId] = useState(defaultOrgId)
+  const [teamName, setTeamName] = useState('')
+  const [teamNameManuallyEdited, setTeamNameManuallyEdited] = useState(false)
+
+  const MAX_TEAM_NAME_LENGTH = 50
+  const generateTeamName = (newUsers: Option[]) => {
+    return newUsers
+      .map((user) => (user.id ? user.label : user.email.split('@')[0]))
+      .join(', ')
+      .substring(0, MAX_TEAM_NAME_LENGTH)
+  }
+
+  const onSelectedUsersChange = (newUsers: Option[]) => {
+    setSelectedUsers(newUsers)
+    const selectedUsersOrganizationIds = new Set()
+    newUsers.forEach((user) => {
+      //add user.organizationIds to selectedUserOrganizationIds
+      user.organizationIds.forEach((organizationId) => {
+        selectedUsersOrganizationIds.add(organizationId)
+      })
+    })
+
+    const mutualOrgs = viewerOrganizations.filter((org) => selectedUsersOrganizationIds.has(org.id))
+
+    const mutualOrgsIds = mutualOrgs.map((org) => org.id)
+    setMutualOrgsIds(mutualOrgsIds)
+    setSelectedOrgId(mutualOrgsIds[0] ?? viewerOrganizations[0]?.id)
+
+    if (!teamNameManuallyEdited) {
+      setTeamName(generateTeamName(newUsers))
+    }
+  }
+
+  const handleAddTeam = () => {
+    const newTeam = {
+      name: teamName.substring(0, MAX_TEAM_NAME_LENGTH),
+      orgId: selectedOrgId
+    }
+    submitMutation()
+
+    AddTeamMutation(
+      atmosphere,
+      {newTeam, invitees: selectedUsers.map((user) => user.email)},
+      {
+        onError,
+        onCompleted: (res, errors) => {
+          onCompleted(res, errors)
+          const error = getGraphQLError(res, errors)
+          if (!error) {
+            onAddTeam(res.addTeam.team.id)
+          }
+        },
+        history,
+        showTeamCreatedToast: false
+      }
+    )
+  }
+
+  const isValid = selectedUsers.length && teamName.trim()
+
+  return (
+    <Dialog isOpen={isOpen} onClose={onClose}>
+      <DialogContent className='z-10'>
+        <DialogTitle className='mb-4'>Add team</DialogTitle>
+
+        <fieldset className={fieldsetStyles}>
+          <label className={labelStyles}>Add teammates</label>
+
+          <AdhocTeamMultiSelect
+            viewerRef={viewer}
+            value={selectedUsers}
+            onChange={onSelectedUsersChange}
+          />
+
+          {selectedUsers.some((user: Option) => !user.id) && (
+            <div className='mt-3 text-xs font-semibold text-slate-700'>
+              Email invitations expire in 30 days.
+            </div>
+          )}
+        </fieldset>
+
+        {showOrgPicker && (
```

Comment:
+1 I think the org picker should always be visible so the height of the modal doesn't change when you select an item. It would also be reassuring for me that I've selected the right team from the right org

Reply:
If org picker will be always visible it will show one single org in most of cases. Would this be ok? Tagging @acressall for this design question

Reply:
I actually don't mind the small height change. I suspect that a lot of users aren't aware of the concept of orgs, and it could be especially confusing in this context, so I only want to bring it up when we actually need the information